tests/tests/test_basic_integration.py

```diff
@@ -139,12 +139,50 @@ def deployment_callback():
                 template = run(catcmd)
                 wait_count = 0
 
+    @pytest.mark.timeout(1000)
+    @pytest.mark.usefixtures("standard_setup_one_client_bootstrapped")
+    def test_forced_inventory_update_from_client(self):
+        """Forces an inventory update from an idling client."""
+        if not env.host_string:
+            execute(self.test_forced_inventory_update_from_clien,
+                    hosts=get_mender_clients())
+            return
+
+        # Give the image a really large wait interval.
+        sedcmd = "sed -i.bak 's/%s/%s/' /etc/mender/mender.conf" % ("\(.*PollInter.*:\)\( *[0-9]*\)", "\\1 1800")
+        out = run(sedcmd)
+        if out.return_code != 0:
+            logger.error(out)
+            pytest.fail("failed to set a large polling interval for the client.")
+        run("systemctl restart mender")
+
+        def deployment_callback():
+            logger.info("Running pre deployment callback function")
+            wait_count = 0
+            # Match the log template six times to make sure the client is truly sleeping.
+            catcmd = "journalctl -u mender --output=cat"
+            template = run(catcmd)
+            while True:
+                logger.info("sleeping...")
+                logger.info("wait_count: %d" % wait_count)
+                time.sleep(10)
+                out = run(catcmd)
+                if out == template:
+                    wait_count += 1
+                    # Only return if the client has been idling in check-wait for a minute.
+                    if wait_count == 6:
+                        return
+                    continue
+                # Update the matching template
+                template = run(catcmd)
+                wait_count = 0
+
         def deployment_triggered_callback():
-            output = run("mender --check-update")
+            output = run("mender --inventory-update")
             if output.return_code != 0:
                 logger.error(output)
-                pytest.fail("Forcing the update check failed")
-            logger.info("mender client has forced an update check")
+                pytest.fail("Forcing the inventory udpate failed")
+            logger.info("mender client has forced an inventory update")
```

Comment:
You ended up removing `deployment_triggered_callback` and the `update_image_successful` call from the already existing `test_forced_update_check_from_client`, which probably was not the intention.

More generally I don't really see how this test is supposed to work. Why do you have to trigger an artifact update to send inventory? I also think the pass condition is too weak, it only checks return code, and does not check that anything actually worked. We should separate better between this test case and the previous one (update check), and check the log and make sure that the forced check was carried out, and at the same time, make sure that the other check was *not* carried out.

Reply:
True. I'm sorry. Was a bit too quick on this one.